### ISSUE TITLE
Configuration variables do not work in unified-builtins2

### ISSUE THREAD

**Issue by andreiarusoaie** [kframework/k-legacy#624]
Example:

```
// Copyright (c) 2010-2014 K Team. All Rights Reserved.
module IMP-SYNTAX
  syntax AExp  ::= "read" "(" ")"
endmodule

module ISSUE
  imports IMP-SYNTAX
  syntax KResult ::= Int | Bool | String

  configuration <T color="yellow">
                  <k color="green"> $PGM:AExp </k>
                  <in color="magenta" stream="stdin"> $IN:List </in>
                </T>

  rule <k> read() => I ...</k>
       <in> ListItem(I:Int) => .List ...</in>  [read]
endmodule
```

On branch master it works like a charm:

```
~/work/k$ git pull
Already up-to-date.
~/work/k$ kompile --version
K-framework nightly build.
Git Revision: 638b554
Build date: Tue Jun 17 21:57:43 EEST 2014
~/work/k$ ant clean build
Buildfile: /Users/andrei/work/k/build.xml
...
~/work/k$ kompile issue.k 
~/work/k$ 
```

On unified builtins it throws an exception:

```
~/work/k$ git checkout unified-builtins2
Switched to branch 'unified-builtins2'
Your branch is up-to-date with 'origin/unified-builtins2'.
~/work/k$ git pull
Already up-to-date.
~/work/k$ ant clean build
Buildfile: /Users/andrei/work/k/build.xml
...
~/work/k$ kompile issue.k 
Exception in thread "main" java.lang.AssertionError: left terms no longer allowed here
    at org.kframework.kil.ListBuiltin.of(ListBuiltin.java:70)
    at org.kframework.kil.DataStructureBuiltin.of(DataStructureBuiltin.java:175)
    at org.kframework.compile.utils.CompileDataStructures.visit(CompileDataStructures.java:145)
    at org.kframework.compile.utils.CompileDataStructures.visit(CompileDataStructures.java:38)
    at org.kframework.kil.KApp.accept(KApp.java:180)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:84)
    at org.kframework.kil.AbstractVisitor.genericVisitChild(AbstractVisitor.java:113)
    at org.kframework.kil.AbstractVisitor.visit(AbstractVisitor.java:421)
    at org.kframework.kil.Cell.accept(Cell.java:287)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:84)
    at org.kframework.kil.AbstractVisitor.genericVisitList(AbstractVisitor.java:135)
    at org.kframework.kil.AbstractVisitor.visit(AbstractVisitor.java:427)
    at org.kframework.kil.AbstractVisitor.visit(AbstractVisitor.java:438)
    at org.kframework.kil.Bag.accept(Bag.java:69)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:84)
    at org.kframework.kil.AbstractVisitor.genericVisitChild(AbstractVisitor.java:113)
    at org.kframework.kil.AbstractVisitor.visit(AbstractVisitor.java:421)
    at org.kframework.kil.Cell.accept(Cell.java:287)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:84)
    at org.kframework.kil.AbstractVisitor.genericVisitChild(AbstractVisitor.java:113)
    at org.kframework.kil.AbstractVisitor.visit(AbstractVisitor.java:421)
    at org.kframework.kil.Cell.accept(Cell.java:287)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:84)
    at org.kframework.kil.AbstractVisitor.genericVisitChild(AbstractVisitor.java:113)
    at org.kframework.kil.AbstractVisitor.visit(AbstractVisitor.java:326)
    at org.kframework.kil.AbstractVisitor.visit(AbstractVisitor.java:334)
    at org.kframework.kil.Configuration.accept(Configuration.java:39)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:84)
    at org.kframework.kil.AbstractVisitor.genericVisitList(AbstractVisitor.java:135)
    at org.kframework.kil.AbstractVisitor.visit(AbstractVisitor.java:300)
    at org.kframework.kil.Module.accept(Module.java:96)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:84)
    at org.kframework.kil.AbstractVisitor.genericVisitList(AbstractVisitor.java:135)
    at org.kframework.kil.AbstractVisitor.visit(AbstractVisitor.java:284)
    at org.kframework.kil.Definition.accept(Definition.java:185)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.compile.utils.CompilerTransformerStep.compile(CompilerTransformerStep.java:24)
    at org.kframework.compile.utils.CompilerSteps.compile(CompilerSteps.java:44)
    at org.kframework.kompile.KompileFrontEnd.genericCompile(KompileFrontEnd.java:192)
    at org.kframework.kompile.KompileFrontEnd.kompile(KompileFrontEnd.java:161)
    at org.kframework.kompile.KompileFrontEnd.main(KompileFrontEnd.java:72)
    at org.kframework.main.Main.main(Main.java:49)
```

I found this on my symbolic backend first, but then I figured out that there is a general problem with configuration variables.

**Comment by andreiarusoaie** [comment]
Guys is there any chance to fix this issue soon?

**Comment by dwightguth** [comment]
Fixed #661
